homeassistant/components/hydrawise/switch.py

```diff
@@ -78,38 +87,42 @@ class HydrawiseSwitch(HydrawiseEntity, SwitchEntity):
     """A switch implementation for Hydrawise device."""
 
     def __init__(
-        self, data, description: SwitchEntityDescription, default_watering_timer
+        self,
+        *,
+        data: dict[str, Any],
+        coordinator: DataUpdateCoordinator,
+        description: SwitchEntityDescription,
+        default_watering_timer,
     ) -> None:
         """Initialize a switch for Hydrawise device."""
-        super().__init__(data, description)
+        super().__init__(data=data, coordinator=coordinator, description=description)
         self._default_watering_timer = default_watering_timer
 
     def turn_on(self, **kwargs: Any) -> None:
         """Turn the device on."""
         relay_data = self.data["relay"] - 1
         if self.entity_description.key == "manual_watering":
-            self.hass.data[DATA_HYDRAWISE].data.run_zone(
-                self._default_watering_timer, relay_data
-            )
+            self.api.run_zone(self._default_watering_timer, relay_data)
         elif self.entity_description.key == "auto_watering":
-            self.hass.data[DATA_HYDRAWISE].data.suspend_zone(0, relay_data)
+            self.api.suspend_zone(0, relay_data)
 
     def turn_off(self, **kwargs: Any) -> None:
         """Turn the device off."""
         relay_data = self.data["relay"] - 1
         if self.entity_description.key == "manual_watering":
-            self.hass.data[DATA_HYDRAWISE].data.run_zone(0, relay_data)
+            self.api.run_zone(0, relay_data)
         elif self.entity_description.key == "auto_watering":
-            self.hass.data[DATA_HYDRAWISE].data.suspend_zone(365, relay_data)
+            self.api.suspend_zone(365, relay_data)
 
-    def update(self) -> None:
+    @callback
+    def _handle_coordinator_update(self) -> None:
         """Update device state."""
         relay_data = self.data["relay"] - 1
-        mydata = self.hass.data[DATA_HYDRAWISE].data
         LOGGER.debug("Updating Hydrawise switch: %s", self.name)
         if self.entity_description.key == "manual_watering":
-            self._attr_is_on = mydata.relays[relay_data]["timestr"] == "Now"
+            self._attr_is_on = self.api.relays[relay_data]["timestr"] == "Now"
         elif self.entity_description.key == "auto_watering":
-            self._attr_is_on = (mydata.relays[relay_data]["timestr"] != "") and (
-                mydata.relays[relay_data]["timestr"] != "Now"
+            self._attr_is_on = (self.api.relays[relay_data]["timestr"] != "") and (
+                self.api.relays[relay_data]["timestr"] != "Now"
```

Comment:
You can check against a set here:
```suggestion
            self._attr_is_on = self.api.relays[relay_data]["timestr"]) not in {"", "Now"}
```

Reply:
Done.